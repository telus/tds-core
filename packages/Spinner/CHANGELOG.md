# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="2.1.0"></a>
# [2.1.0](https://github.com/telusdigital/tds/compare/@tds/core-spinner@2.0.3...@tds/core-spinner@2.1.0) (2018-12-19)


### Features

* **core-spinner:** add inline property to render as inline-block ([b11ebac](https://github.com/telusdigital/tds/commit/b11ebac))




<a name="2.0.3"></a>
## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-spinner@2.0.2...@tds/core-spinner@2.0.3) (2018-12-06)




**Note:** Version bump only for package @tds/core-spinner

<a name="2.0.2"></a>
## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-spinner@2.0.1...@tds/core-spinner@2.0.2) (2018-07-10)




**Note:** Version bump only for package @tds/core-spinner

<a name="2.0.1"></a>
## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-spinner@2.0.0...@tds/core-spinner@2.0.1) (2018-06-14)




**Note:** Version bump only for package @tds/core-spinner

<a name="2.0.0"></a>
# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-spinner@1.0.0...@tds/core-spinner@2.0.0) (2018-03-26)


### Features

* **core-spinner:** add a11yLabel prop ([2766d48](https://github.com/telusdigital/tds/commit/2766d48))
* **core-spinner:** forward rest props onto the svg, but disallow custom css classes or styles ([a6f9033](https://github.com/telusdigital/tds/commit/a6f9033))
* **core-spinner:** remove full screen mode ([ce637c3](https://github.com/telusdigital/tds/commit/ce637c3)), closes [#439](https://github.com/telusdigital/tds/issues/439)
* **core-spinner:** remove wrapperClassNames prop ([29842e2](https://github.com/telusdigital/tds/commit/29842e2))
* **core-spinner:** start moving styles into css modules ([e38de2b](https://github.com/telusdigital/tds/commit/e38de2b))
* **core-spinner:** switch all classes to css modules ([ac255b3](https://github.com/telusdigital/tds/commit/ac255b3))


### BREAKING CHANGES

* **core-spinner:** removal of wrapperClassNames
* **core-spinner:** removal of the full screen behaviour
