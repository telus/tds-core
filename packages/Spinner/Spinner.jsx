import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { position } from '@tds/shared-styles'

import { warn } from '../../shared/utils/warn'
import safeRest from '../../shared/utils/safeRest'
import SpinnerSvg from './SpinnerSvg/SpinnerSvg'

const zindexModalBackdrop = 1400

const SpinnerContainer = styled.div(({ inline, fullScreen }) => ({
  ...position.relative,
  ...(inline && { display: 'inline-block' }),
  ...(fullScreen && position.centerVertically),
}))

const ContentOverlay = styled.div({
  position: 'absolute',
  width: '100%',
  height: '100%',
  zIndex: zindexModalBackdrop,
})

const FullscreenOverlay = styled.div({
  position: 'fixed',
  width: '100vw',
  height: '100vh',
  top: 0,
  left: 0,
  zIndex: zindexModalBackdrop,
  backgroundColor: 'rgba(255, 255, 255, 0.96)',
})

const OpaqueContainer = styled.div({
  opacity: 0.06,
})

/**
 * A waiting indicator.
 *
 * @version ./package.json
 */

class Spinner extends React.PureComponent {
  constructor() {
    super()
    this.spinnerOverlayRef = null
  }

  componentDidUpdate() {
    if (this.props.fullScreen && this.props.spinning) {
      document.body.addEventListener('touchmove', this.preventScroll, { passive: false })
      document.body.addEventListener('touchstart', this.preventScroll, { passive: false })

      document.body.style.overflow = 'hidden'
    } else {
      document.body.removeEventListener('touchmove', this.preventScroll)
      document.body.removeEventListener('touchstart', this.preventScroll)
      document.body.style.overflow = 'auto'
    }
  }

  preventScroll = e => {
    if (this.spinnerOverlayRef.current.contains(e.targetTouches[0].target)) {
      e.preventDefault()
    }
  }

  render() {
    const {
      spinning,
      label,
      dangerouslyHideVisibleLabel,
      inline,
      size,
      variant,
      fullScreen,
      children,
      ...rest
    } = this.props

    if (size === 'large' && variant === 'secondary') {
      warn(
        'core-spinner',
        'The Spinner should not use the `secondary` variant while `size` is set to `large`.'
      )
    }

    if (!spinning) {
      return children || null
    }

    const spinnerSvg = props => (
      <SpinnerSvg
        {...props}
        tip={dangerouslyHideVisibleLabel || size === 'small' ? undefined : label}
        a11yLabel={label}
        size={size}
        variant={variant}
        {...safeRest(rest)}
      />
    )

    if (fullScreen) {
      return (
        <FullscreenOverlay
          ref={el => {
            this.spinnerOverlayRef = el
          }}
          data-testid="overlay"
        >
          <SpinnerContainer
            inline={inline}
            fullScreen={fullScreen}
            data-testid="container"
            aria-live="assertive"
          >
            {spinnerSvg({ overlay: true })}
          </SpinnerContainer>
        </FullscreenOverlay>
      )
    }
    if (children) {
      return (
        <SpinnerContainer
          inline={inline}
          fullScreen={fullScreen}
          data-testid="container"
          aria-live="assertive"
        >
          {spinnerSvg({ overlay: true })}

          <ContentOverlay data-testid="overlay" />

          <OpaqueContainer
            onFocus={e => {
              e.target.blur()
            }}
            aria-hidden="true"
          >
            {children}
          </OpaqueContainer>
        </SpinnerContainer>
      )
    }

    return spinnerSvg()
  }
}

Spinner.propTypes = {
  /**
   * Whether or not to render the spinner.
   */
  spinning: PropTypes.bool,
  /**
   * Communicates a message to assistive technology while visible. This same message will appear underneath the spinner when its `size` is `large`.
   *
   * When used with `A11yContent`, labe text should be wrapped by a `<span>` or `<React.Fragment>`.
   *
   * @since 2.2.0
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /**
   * Hides the visible label under the spinner when the spinner's `size` is set to `large`. For special circumstances only.
   * @ignore
   */
  dangerouslyHideVisibleLabel: PropTypes.bool,
  /**
   * Render the Spinner as inline-block. This can be used when wrapping
   * interactive elements such as buttons.
   *
   * @since 2.1.0
   */
  inline: PropTypes.bool,
  /**
   * The size of the spinner
   *
   * @since 2.2.0
   */
  size: PropTypes.oneOf(['large', 'small']),
  /**
   * The spinner's colour. The `secondary` variant is only available when `size` is set to `small`.
   *
   * @since 2.2.0
   */
  variant: PropTypes.oneOf(['primary', 'secondary']),
  /**
   * Enables body locking
   *
   * @since 2.2.0
   */
  fullScreen: PropTypes.bool,
  /**
   * Content to be overlaid while the spinner is active. Can be text, any HTML element,
   * or any component.
   */
  children: PropTypes.node,
}

Spinner.defaultProps = {
  spinning: false,
  dangerouslyHideVisibleLabel: false,
  inline: false,
  size: 'large',
  variant: 'primary',
  fullScreen: false,
  children: undefined,
}

export default Spinner
