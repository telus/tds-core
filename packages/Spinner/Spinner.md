### Usage criteria

- Avoid using a spinner whenever possible. Make content available to the user as soon as possible
- User feedback should be present when a user action takes longer than 0.1 seconds
- A Spinner is most appropriate for providing feedback for actions that last between 2 and 10 seconds
- Actions that take longer than 10 seconds ideally provide feedback with relative progress, such as a progress bar
- Recommend not to use for full page loading
- Label should be short and concise. Spinner label should accurately explain the state of the requested action along with relevant items being loaded, "Loading page content", "Logging in", "Processing payment"
- Use the small spinner when applied on a button or other small interactive elements such as toggles or links
  - When overlaying buttons, the spinner’s colour should match the button's colour (green spinner for green buttons, and purple spinner for purple buttons)
- Use the large spinner when affecting an entire web page or content block, whose content has not finished loading, such as paginated content
- Both the small and large spinner have an overlay background to visually indicate content behind it is inaccessible.
  Use the standalone spinner (small/large) as an interstitial placeholder for loading content

### Minimal usage

- To make the `Spinner` visible, set the `spinning` prop to `true`
- Provide a `label` to give more context about what is happening
  - The `label` prop is a `string` or `node` that acts as both a line of text under the `Spinner` as well as text that is communicated to assistive technology
  - Use `A11yContent` to provide information to assistive technology that should not appear in the `Spinner`'s visible label. This content must be wrapped with a `React.Fragment` or `span` if `A11yContent` is being used. In the next major update, the `label` prop will be required in code

#### Standard `label`

```jsx
<Spinner spinning label="Loading addresses" />
```

#### `label` with `A11yContent`

```jsx
<Spinner
  spinning
  label={
    <>
      Loading <A11yContent>the address list. Please wait.</A11yContent>
    </>
  }
/>
```

### Overlaying the spinner and the `inline` prop

- The spinner can be overlaid on a section of content, which will prevent interactions with the content while active
  - Please note, blocking interaction is often not a desirable user experience
- Avoid overlaying the entire window with the `Spinner` when possible
- Wrap the `Spinner` around the content to overlay it
  - If the `Spinner` should only cover its children, set the `inline` prop to `true`. If the `Spinner` should cover the full width of its parent regardless of the size of its children, `inline` should be set to `false`

#### Overlaying buttons

- When overlaying a TDS `Button`, the `Spinner` must have a `size` of `small`
- The `Spinner`'s `variant` must match that of the the `Button`
- The `Spinner` should have `inline` set to `true` to fully cover each individual button

```jsx
<Box between={2}>
  <Box>
    <Spinner spinning size="small" label="Request is processing." inline>
      <Button>Login</Button>
    </Spinner>
  </Box>
  <Box>
    <Spinner spinning size="small" variant="secondary" label="Request is processing." inline>
      <Button variant="secondary">Login</Button>
    </Spinner>
  </Box>
</Box>
```

#### Overlaying all other content

- When overlaying general page content, the `Spinner` must have its `size` set to `large` and `variant` set to `primary`
- A `label` should also be provided, as it is required for using the large `Spinner`

```jsx
initialState = {
  contentLoading: false,
}

const loadContent = () => {
  setState({ contentLoading: true })

  setTimeout(() => setState({ contentLoading: false }), 2000)
}

;<Box between={2}>
  <Spinner
    label={
      <span>
        Loading <A11yContent>the page content.</A11yContent>
      </span>
    }
    spinning={state.contentLoading}
    inline
  >
    <Card>
      <Box between={4}>
        <Box between={3}>
          <Heading level="h2">Availability Check</Heading>
          <Paragraph>Ready to order? Click here to check if the product is available.</Paragraph>
        </Box>
        ​<div>
          <Button onClick={loadContent}>Check Availability</Button>
        </div>
      </Box>
    </Card>
  </Spinner>
</Box>
```

### Accessibility and focus

#### Overlaying a button

- When overlaying a single button, a container with `aira-live` set to `assertive` should be used _near the overlaid Button_
- When the process is complete, `A11yContent` can be used to trigger a message indicating that the task has been completed
- The added content must be close to the spinner's location when it appears
  - For example, do not have a spinner from the main body modify content in the page's footer
- When the content is done loading, focus on the content to allow magnifiers to move to it

```jsx
initialState = {
  contentLoading: false,
  loadedMessage: '',
  loadedContent: '',
}

const loadContent = () => {
  setState({ contentLoading: true })

  setTimeout(
    () =>
      setState({
        contentLoading: false,
        loadedMessage: 'Product stock found',
        loadedContent: '5 products available',
      }),
    2000
  )
}

;<Box between={2}>
  <Card>
    <Box between={4}>
      <Box between={3}>
        <Heading level="h2">Availability Check</Heading>
        <Paragraph>Ready to order? Click here to check if the product is available.</Paragraph>
      </Box>
      ​<div>
        <Spinner label="Loading address" size="small" spinning={state.contentLoading} inline>
          <Button onClick={loadContent}>Check Availability</Button>
        </Spinner>
      </div>
      <div aria-live="assertive">
        <A11yContent>{state.loadedMessage}</A11yContent>
        {state.loadedContent}
      </div>
    </Box>
  </Card>
</Box>
```

#### Overlaying a full layout

Overlaying a full layout is similar to overlaying a single Button. The same methods to achieve accessibility apply, including the important rule of having content that is the result of the spinner appear close to the spinner when complete. Visually, the spinner `size` must be set to `large` and the `variant` set to `primary`.

```jsx
initialState = {
  contentLoading: false,
  loadedMessage: '',
  loadedContent: '',
}

const loadContent = () => {
  setState({ contentLoading: true })

  setTimeout(
    () =>
      setState({
        contentLoading: false,
        loadedMessage: 'Products found',
        loadedContent: (
          <UnorderedList>
            <UnorderedList.Item>iPhone XS</UnorderedList.Item>
            <UnorderedList.Item>Galaxy S10</UnorderedList.Item>
            <UnorderedList.Item>Pixel 3</UnorderedList.Item>
          </UnorderedList>
        ),
      }),
    2000
  )
}

;<Box between={2}>
  <Heading level="h2">Available products:</Heading>
  <Card>
    <Spinner label="Finding products" spinning={state.contentLoading}>
      <Box between={3} vertical={2}>
        <div>
          <Button onClick={loadContent}>Check Availability</Button>
        </div>
        <div aria-live="assertive">
          <A11yContent>{state.loadedMessage}</A11yContent>
          {state.loadedContent}
        </div>
      </Box>
    </Spinner>
  </Card>
</Box>
```

### Displaying a full screen spinner

To block the entire screen while waiting, do not wrap any children and use the `fullScreen` prop. This will disable scrolling and prevent any interactions with the page while it is active. It is recommended to avoid this pattern where possible, as this takes control completely away from the user.

```jsx
initialState = {
  contentLoading: false,
  loadedMessage: '',
  loadedContent: <Paragraph>Page 1</Paragraph>,
}

const loadContent = () => {
  setState({ contentLoading: true })

  setTimeout(
    () =>
      setState({
        contentLoading: false,
        loadedMessage: 'Page 2 Loaded',
        loadedContent: <Paragraph>Page 2</Paragraph>,
      }),
    2000
  )
}

;<Box between={2}>
  <Card>
    <Spinner
      fullScreen={state.contentLoading}
      label="Loading page"
      spinning={state.contentLoading}
    />
    <Box between={3} vertical={2}>
      <div>
        <Button onClick={loadContent}>Load Page 2</Button>
      </div>
      <div aria-live="assertive">
        <A11yContent>{state.loadedMessage}</A11yContent>
        {state.loadedContent}
      </div>
    </Box>
  </Card>
</Box>
```
