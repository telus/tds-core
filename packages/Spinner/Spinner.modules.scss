$zindex-modal-backdrop: 1400;

.overlay {
  position: absolute;
  width: 100%;
  height: 100%;

  z-index: $zindex-modal-backdrop;
}

.fullscreenOverlay {
  position: fixed;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;

  z-index: $zindex-modal-backdrop;
  background-color: rgba(255, 255, 255, 0.96);
}

.opaque {
  opacity: 0.06;
}

.inline {
  display: inline-block;
}
