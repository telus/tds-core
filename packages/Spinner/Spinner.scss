@import '~@tds/core-colours/colours';
@import '../../shared/scss/responsive';

$zindex-negative: -1 !default;
$zindex-modal-backdrop: 1400 !default;

.spinner-wrapper {
  position: relative;

  .spinner {
    position: absolute;
    top: calc(50% - 50px);
    left: 0;
    width: 100%;
  }
}

.spinner-container {
  transition: opacity 0.3s;

  &--loading {
    opacity: 0.1;
    z-index: $zindex-negative;
  }
}

.spinner-wrapper__content-blocker {
  z-index: $zindex-modal-backdrop;
  position: absolute;
  width: 100%;
  height: 100%;
}
