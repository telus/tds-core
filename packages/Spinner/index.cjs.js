const Spinner = require('./dist/index.cjs')

module.exports = Spinner
