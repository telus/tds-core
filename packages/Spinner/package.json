{
  "name": "@tds/core-spinner",
  "version": "1.0.0",
  "description": "A waiting indicator",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "@tds/core-text": "^1.0.0",
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "@tds/core-colours": "^1.0.0",
    "classnames": "^2.2.5"
  }
}
