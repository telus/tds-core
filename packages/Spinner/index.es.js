import Spinner from './dist/index.es'

export default Spinner
