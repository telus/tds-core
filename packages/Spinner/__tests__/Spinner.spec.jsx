import React from 'react'
import { shallow, mount, render } from 'enzyme'

import Text from '@tds/core-text'

import Spinner from '../Spinner'

describe('Spinner', () => {
  const defaultProps = {
    label: 'Loading content',
    spinning: true,
  }
  const doShallow = (overrides = {}, children) =>
    shallow(
      <Spinner {...defaultProps} {...overrides}>
        {children}
      </Spinner>
    )

  const doMount = (overrides = {}, children) => {
    const spinner = mount(
      <Spinner {...defaultProps} {...overrides}>
        {children}
      </Spinner>
    )

    const spinnerSvg = spinner.find('[data-testid="svg"]')

    return {
      spinner,
      container: spinner.find('[data-testid="container"]'),
      overlay: spinner.find('[data-testid="overlay"]'),
      spinnerContainer: spinner.find('[data-testid="spinner"]'),
      spinnerSvg,
      a11yLabel: spinnerSvg.find('title[id^="spinner-title-"]'),
    }
  }

  it('renders', () => {
    const spinner = render(<Spinner label="Loading content" spinning />)

    expect(spinner).toMatchSnapshot()
  })

  it('renders with children', () => {
    const spinner = render(
      <Spinner label="Loading content" spinning>
        <span>Overlay me with the spinner</span>
      </Spinner>
    )

    expect(spinner).toMatchSnapshot()
  })

  it('renders a small spinner', () => {
    const { container } = doMount(
      { size: 'small', inline: true },
      <span>Overlay me with the spinner</span>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders a small secondary spinner', () => {
    const { container } = doMount(
      { size: 'small', variant: 'secondary', inline: true },
      <span>Overlay me with the spinner</span>
    )
    expect(container).toMatchSnapshot()
  })

  it('does not render a large secondary spinner', () => {
    const { container } = doMount(
      { size: 'large', variant: 'secondary', inline: true },
      <span>Overlay me with the spinner</span>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders inline', () => {
    const { container } = doMount(
      { spinning: true, inline: true },
      <span>Overlay me with the spinner</span>
    )
    expect(container).toHaveClassName('inline')
  })

  it('is only visible while spinning', () => {
    let spinner = doShallow({ spinning: false })
    expect(spinner).toBeEmptyRender()

    spinner = doShallow({ spinning: true })
    expect(spinner).not.toBeEmptyRender()
  })

  it('can have a tip', () => {
    const { spinner } = doMount({ label: undefined, tip: 'A tip' })

    expect(spinner).toContainReact(<Text size="small">A tip</Text>)
  })

  it('can have a label', () => {
    const { spinner } = doMount({ label: 'A label' })

    expect(spinner).toContainReact(<Text size="small">A label</Text>)
  })

  describe('overlaying content', () => {
    it('places the spinner on top of the content while spinning', () => {
      const { overlay, spinnerContainer } = doMount(
        { spinning: true },
        <span>Overlay me with the spinner</span>
      )

      expect(overlay).toExist()
      expect(spinnerContainer).toHaveClassName('centered')
    })

    it('only shows the children while not spinning', () => {
      const spinner = doShallow({ spinning: false }, <span>Overlay me with the spinner</span>)

      expect(spinner).toMatchElement(<span>Overlay me with the spinner</span>)
    })

    it('can be fullscreen', () => {
      const spinner = doShallow({ fullScreen: true, label: 'Loading page', spinning: true })

      expect(spinner).toHaveClassName('fullscreenOverlay')
    })
  })

  describe('accessibility', () => {
    it('gives the svg a default a11y label', () => {
      const { spinnerSvg, a11yLabel } = doMount({ label: undefined })

      expect(a11yLabel).toHaveText(
        'A spinner is active. Please wait while the page completes a task.'
      )
      expect(spinnerSvg).toHaveProp('aria-labelledby', a11yLabel.props().id)
    })

    it('allows a custom a11y label (a11yLabel)', () => {
      const { a11yLabel } = doMount({ label: undefined, a11yLabel: 'Something is busy' })

      expect(a11yLabel).toHaveText('Something is busy')
    })

    it('allows a custom a11y label (label)', () => {
      const { a11yLabel } = doMount({ label: 'Something is busy' })

      expect(a11yLabel).toHaveText('Something is busy')
    })
  })

  it('passes additional attributes to svg element', () => {
    const { spinnerSvg } = doMount({ id: 'the-spinner', 'data-some-attr': 'some value' })

    expect(spinnerSvg).toHaveProp({ id: 'the-spinner', 'data-some-attr': 'some value' })
  })

  it('does not allow custom CSS', () => {
    const { spinnerSvg } = doMount({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(spinnerSvg).not.toHaveProp('className', 'my-custom-class')
    expect(spinnerSvg).not.toHaveProp('style')
  })
})
