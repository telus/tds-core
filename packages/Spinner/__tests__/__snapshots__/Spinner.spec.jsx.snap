// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner does not render a large secondary spinner 1`] = `
<div
  aria-live="assertive"
  className="relative inline"
  data-testid="container"
>
  <SpinnerSvg
    a11yLabel="Loading content"
    overlay={true}
    size="large"
    tip="Loading content"
    variant="secondary"
  >
    <div
      className="container centered"
      data-testid="spinner"
    >
      <svg
        aria-labelledby="spinner-title-5"
        aria-live="assertive"
        className="svg"
        data-testid="svg"
        height="100"
        role="alert"
        viewBox="0 0 100 100"
        width="100"
      >
        <title
          id="spinner-title-5"
        >
          Loading content
        </title>
        <circle
          className="primaryCircle"
          cx="50"
          cy="50"
          fill="none"
          r="20"
          strokeDasharray="89, 200"
          strokeDashoffset="0"
          strokeLinecap="round"
          strokeWidth="4"
        />
      </svg>
      <div
        className="tip"
      >
        <Text
          block={false}
          bold={false}
          invert={false}
          size="small"
        >
          <span
            className="TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__smallFont___3S6nz TDS_Typography-modules__color___2CNH8"
          >
            Loading content
          </span>
        </Text>
      </div>
    </div>
  </SpinnerSvg>
  <div
    className="overlay"
    data-testid="overlay"
  />
  <div
    aira-hidden="true"
    className="opaque"
    onFocus={[Function]}
  >
    <span>
      Overlay me with the spinner
    </span>
  </div>
</div>
`;

exports[`Spinner renders 1`] = `
<div
  class="container"
  data-testid="spinner"
>
  <svg
    aria-labelledby="spinner-title-1"
    aria-live="assertive"
    class="svg"
    data-testid="svg"
    height="100"
    role="alert"
    viewBox="0 0 100 100"
    width="100"
  >
    <title
      id="spinner-title-1"
    >
      Loading content
    </title>
    <circle
      class="primaryCircle"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      stroke-dasharray="89, 200"
      stroke-dashoffset="0"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <div
    class="tip"
  >
    <span
      class="TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__smallFont___3S6nz TDS_Typography-modules__color___2CNH8"
    >
      Loading content
    </span>
  </div>
</div>
`;

exports[`Spinner renders a small secondary spinner 1`] = `
<div
  aria-live="assertive"
  className="relative inline"
  data-testid="container"
>
  <SpinnerSvg
    a11yLabel="Loading content"
    overlay={true}
    size="small"
    variant="secondary"
  >
    <div
      className="container centered"
      data-testid="spinner"
    >
      <svg
        aria-labelledby="spinner-title-4"
        aria-live="assertive"
        className="svg"
        data-testid="svg"
        height="50"
        role="alert"
        viewBox="0 0 100 100"
        width="50"
      >
        <title
          id="spinner-title-4"
        >
          Loading content
        </title>
        <circle
          className="secondaryCircle"
          cx="50"
          cy="50"
          fill="none"
          r="20"
          strokeDasharray="89, 200"
          strokeDashoffset="0"
          strokeLinecap="round"
          strokeWidth="4"
        />
      </svg>
    </div>
  </SpinnerSvg>
  <div
    className="overlay"
    data-testid="overlay"
  />
  <div
    aira-hidden="true"
    className="opaque"
    onFocus={[Function]}
  >
    <span>
      Overlay me with the spinner
    </span>
  </div>
</div>
`;

exports[`Spinner renders a small spinner 1`] = `
<div
  aria-live="assertive"
  className="relative inline"
  data-testid="container"
>
  <SpinnerSvg
    a11yLabel="Loading content"
    overlay={true}
    size="small"
    variant="primary"
  >
    <div
      className="container centered"
      data-testid="spinner"
    >
      <svg
        aria-labelledby="spinner-title-3"
        aria-live="assertive"
        className="svg"
        data-testid="svg"
        height="50"
        role="alert"
        viewBox="0 0 100 100"
        width="50"
      >
        <title
          id="spinner-title-3"
        >
          Loading content
        </title>
        <circle
          className="primaryCircle"
          cx="50"
          cy="50"
          fill="none"
          r="20"
          strokeDasharray="89, 200"
          strokeDashoffset="0"
          strokeLinecap="round"
          strokeWidth="4"
        />
      </svg>
    </div>
  </SpinnerSvg>
  <div
    className="overlay"
    data-testid="overlay"
  />
  <div
    aira-hidden="true"
    className="opaque"
    onFocus={[Function]}
  >
    <span>
      Overlay me with the spinner
    </span>
  </div>
</div>
`;

exports[`Spinner renders with children 1`] = `
<div
  aria-live="assertive"
  class="relative"
  data-testid="container"
>
  <div
    class="container centered"
    data-testid="spinner"
  >
    <svg
      aria-labelledby="spinner-title-2"
      aria-live="assertive"
      class="svg"
      data-testid="svg"
      height="100"
      role="alert"
      viewBox="0 0 100 100"
      width="100"
    >
      <title
        id="spinner-title-2"
      >
        Loading content
      </title>
      <circle
        class="primaryCircle"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        stroke-dasharray="89, 200"
        stroke-dashoffset="0"
        stroke-linecap="round"
        stroke-width="4"
      />
    </svg>
    <div
      class="tip"
    >
      <span
        class="TDS_Typography-modules__small___mt5lS TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__smallFont___3S6nz TDS_Typography-modules__color___2CNH8"
      >
        Loading content
      </span>
    </div>
  </div>
  <div
    class="overlay"
    data-testid="overlay"
  />
  <div
    aira-hidden="true"
    class="opaque"
  >
    <span>
      Overlay me with the spinner
    </span>
  </div>
</div>
`;
